lib/src/io_utils.dart

```diff
@@ -19,15 +25,73 @@ String resolveTildePath(String originalPath) {
 
   String homeDir;
 
-  if (Platform.isWindows) {
-    homeDir = path.absolute(Platform.environment['USERPROFILE']);
+  if (io.Platform.isWindows) {
+    homeDir = path.absolute(io.Platform.environment['USERPROFILE']);
   } else {
-    homeDir = path.absolute(Platform.environment['HOME']);
+    homeDir = path.absolute(io.Platform.environment['HOME']);
   }
 
   return path.join(homeDir, originalPath.substring(2));
 }
 
+extension ResourceProviderExtensions on ResourceProvider {
+  Folder createSystemTemp(String prefix) {
+    if (this is PhysicalResourceProvider) {
+      return getFolder(io.Directory.systemTemp.createTempSync(prefix).path);
+    } else {
+      return getFolder(pathContext.join('/tmp', prefix))..create();
+    }
+  }
+
+  Folder get defaultSdkDir {
+    if (this is PhysicalResourceProvider) {
+      var sdkDir = getFile(pathContext.absolute(io.Platform.resolvedExecutable))
+          .parent
+          .parent;
+      assert(pathContext.equals(
+          sdkDir.path, PubPackageMeta.sdkDirParent(sdkDir, this).path));
+      return sdkDir;
+    } else {
+      return getFolder('TODO(srawlins)');
+    }
+  }
+
+  String get resolvedExecutable {
+    if (this is PhysicalResourceProvider) {
+      return io.Platform.resolvedExecutable;
+    } else {
+      return 'TODO(srawlins)';
```

Comment:
I assume this PR is still a WIP?

Reply:
Sorry, that part is not implemented yet. It will be the part used in tests with MemoryResourceProvider. Moved TODO to comment; return null.